Source/RapyutaSimulationPlugins/Public/Core/RRBaseActor.h

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020-2021 Rapyuta Robotics Co., Ltd.
+
+#pragma once
+
+// std
+#include <mutex>
+
+// UE
+#include "GameFramework/Pawn.h"
+
+// RapyutaSimulationPlugins
+#include "Core/RRActorCommon.h"
+
+#include "RRBaseActor.generated.h"
+
+class ARRGameMode;
+class ARRGameState;
+class ARRPlayerController;
+
+/* Base actor class for all Rapyuta Sim actors:
+ * - Be assigned a sim mode instance id
+ * - Provide accessible handles to the common Game framework objects (GameInstance, GameMode, GameState) and
+ * the corresponding [ActorCommon] of the Sim Mode instance it belongs to.
+ * - Does not necessarily has a USceneComponent as Root!
```

Comment:
can you write down the merit comes from not having a USceneCompoennt.

Reply:
Sure, SceneComponent deals with object poses and their accompanying specific transform calculations. Also, in the hierarchy, `USceneComponent` is a child class of `UActorComponent`, so factually the parent classes are less expensive than child classes instances. 
Since this is the base actor class for all in the Sim, it could parent both mesh-based and non-meshed based ones (like the SimulationState), it should be as lean as possible.

Reply:
Sure, SceneComponent deals with object poses and their accompanying specific transform calculations. Also, in the hierarchy, `USceneComponent` is a child class of `UActorComponent`, so factually the parent classes are less expensive than child ones. 
Since this is the base actor class for all in the Sim, it could parent both mesh-based and non-meshed based ones (like the SimulationState), it should be as lean as possible.